### PR TITLE
Get service endpoints from ledger metadata

### PR DESCRIPTION
**Pull request by emschwartz**
To be more flexible and RESTful the plugin should use the `urls` returned in the ledger metadata instead of assuming where the ledger endpoints are hosted relative to the base URL.

Thanks to @bensharaf for refactoring the tests into separate files. This makes it easier to add new tests.

Resolves https://github.com/interledger/js-ilp-plugin-bells/issues/48

### CODE REVIEW

File: src/lib/plugin.js [hunk 1]
```diff
@@ -130,7 +131,7 @@ class FiveBellsLedger extends EventEmitter2 {
 
     // Resolve ledger metadata
     const ledgerMetadata = yield this._fetchLedgerMetadata()
-    this.accountUriTemplate = ledgerMetadata.urls.account
+    this.urls = ledgerMetadata.urls
```
Comment (justmoon) [inline src/lib/plugin.js]: You should sanitize this object, before making it part of the plugin's state. Something like this:

``` js
this.urls = {
  account_transfers: ledgerMetadata.account_transfers,
  //...
}

// Check required ledger endpoints are present
if (typeof this.urls.account_transfers !== 'string') {
  throw new Error('...')
}
```

It's better to fail when connecting if we can than at some random later time.

File: test/connectSpec.js [hunk 2]
```diff
@@ -110,6 +110,37 @@ describe('Connection methods', function () {
       infoNock.done()
     })
 
+    it('should set urls from object in metadata', function * () {
```
Comment (justmoon) [inline test/connectSpec.js]: Should also test that no unexpected extra properties make it into `plugin.urls`.

File: test/connectSpec.js [hunk 3]
```diff
@@ -228,14 +256,14 @@ describe('Connection methods', function () {
 
   describe('disconnect', function () {
     beforeEach(function * () {
-      const nockAccount = nock('http://red.example')
+      nock('http://red.example')
```
Comment (justmoon) [inline test/connectSpec.js:443]: Should assert that these nocks are done on `afterEach`.

**Review by justmoon — CHANGES_REQUESTED** [review]

File: test/connectSpec.js (hunk 3 above)
Comment (emschwartz) [reply test/connectSpec.js:443]: See https://github.com/interledger/js-ilp-plugin-bells/blob/urls-from-metadata/test/connectSpec.js#L33

File: test/connectSpec.js (hunk 2 above)
Comment (emschwartz) [reply test/connectSpec.js]: Fixed in https://github.com/interledger/js-ilp-plugin-bells/pull/52/commits/71ee1b7e82db7817f3be32da6345943605cb7de8

File: src/lib/plugin.js (hunk 1 above)
Comment (emschwartz) [reply src/lib/plugin.js]: Addressed in https://github.com/interledger/js-ilp-plugin-bells/pull/52/commits/71ee1b7e82db7817f3be32da6345943605cb7de8

File: test/connectSpec.js [hunk 4]
```diff
@@ -283,3 +481,17 @@ function * assertResolve (promise, expected) {
   assert(promise instanceof Promise)
   assert.deepEqual(yield promise, expected)
 }
+
+function * assertRejectsWith (promise, errorConstructor, errorMessage) {
```
Comment (justmoon) [inline test/connectSpec.js]: Use `chai-as-promised`.

**Review by justmoon — CHANGES_REQUESTED** [review]

File: test/connectSpec.js (hunk 4 above)
Comment (emschwartz) [reply test/connectSpec.js]: Addressed in https://github.com/interledger/js-ilp-plugin-bells/pull/52/commits/24328b6e34c5f570f1b0c224861b8fcf3ac15e14

**Review by justmoon — APPROVED** [review]

**Pull request merged**